src/nni_manager/training_service/kubernetes/frameworkcontroller/frameworkcontrollerConfig.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+'use strict';
+
+import { KubernetesClusterConfig, KubernetesStorageKind, NFSConfig, AzureStorage, keyVaultConfig, KubernetesTrialConfig, KubernetesTrialConfigTemplate } from '../kubernetesConfig'
+
+export class FrameworkControllerClusterConfigNFS extends KubernetesClusterConfig{
+    public readonly nfs: NFSConfig;
+    
+    constructor(nfs: NFSConfig, storage?: KubernetesStorageKind) {
+        super(storage)
+        this.nfs = nfs;
+    }
+
+    public get storageType(): KubernetesStorageKind{
+        return 'nfs';
+    }
+}
+
+export class FrameworkControllerClusterConfigAzure extends KubernetesClusterConfig{
+    public readonly keyVault: keyVaultConfig;
+    public readonly azureStorage: AzureStorage;
+    
+    constructor(keyVault: keyVaultConfig, azureStorage: AzureStorage, storage?: KubernetesStorageKind) {
+        super(storage)
+        this.keyVault = keyVault;
+        this.azureStorage = azureStorage;
+    }
+
+    public get storageType(): KubernetesStorageKind{
+        return 'azureStorage';
+    }
+}
+
+export class FrameworkControllerClusterConfigFactory {
+
+    public static generateFrameworkControllerClusterConfig(jsonObject: object): KubernetesClusterConfig {
+         let frameworkcontrollerClusterConfigObject = <KubernetesClusterConfig>jsonObject;
+         if(frameworkcontrollerClusterConfigObject.storage && frameworkcontrollerClusterConfigObject.storage === 'azureStorage') {
```

Comment:
This method could be refactorred as:

<generatStorage>jsonObject;
        switch(generalStorage.storage) {
            case 'nfs':
                NFSConfig.generateInance(object);
            case 'azure':
                aZURECONFIG.generateInstance(object);
        }

you can define static generateInsance method in each sub-classes.